python/pyxet/pyxet/file_system.py

```diff
@@ -8,17 +8,51 @@
 
 from .commit_transaction import MultiCommitTransaction
 from .file_interface import XetFile
-from .url_parsing import parse_url, XetPathInfo
+from .url_parsing import parse_url, XetPathInfo, normalize_domain, set_default_domain
 
 if 'SPHINX_BUILD' not in os.environ:
     from .rpyxet import rpyxet
-    _manager = rpyxet.PyRepoManager()
+
+__repo_managers = {}
+__login_credentials = {}
+
+def _repo_manager(domain):
+    global __repo_managers
+    global __login_credentials
+
+    domain = normalize_domain(domain)
+
+    try:
+        return __repo_managers[domain]
+    except KeyError:
+        pass
+
+    repo = rpyxet.PyRepoManager(domain)
+    __login_credentials
+    if domain in __login_credentials:
+        repo.override_login_config(*__login_credentials[domain])
+    elif None in __login_credentials:
+        repo.override_login_config(*__login_credentials[None])
+
+    __repo_managers[domain] = repo
+    return repo
+
 
 def login(user, token, email=None, host=None):
     """
     Sets the active login credentials used to authenticate against Xethub.
     """
-    _manager.override_login_config(user, token, email, host)
+    global __login_credentials
+    if host is not None:
+        host = normalize_domain(host)
+        set_default_domain(host)
+    __login_credentials[host] = (user, token, email)
```

Comment:
What's the purpose of saving this in a memory cache? It's another layer of consistency we need to maintain (like interaction with invoking `git xet login`).

Reply:
Because we now have to instantiate the XetRepoManager lazily once we know the endpoint, `git xet login` can be run beforehand so we need to have it hang around so that we can set up XetRepoManager appropriately.

Reply:
Does XetRepoManager not pick up credentials from .xetconfig?

Reply:
Discussed offline, this is actually needed.